Web Inspector: drop support for iOS 12.* and macOS 12.*

dcrousso opened the pull request:
#### 1eab38a566c6d7289d2387edc9bb291404649867
<pre>
Web Inspector: drop support for iOS 12.* and macOS 12.*
<a href="https://bugs.webkit.org/show_bug.cgi?id=282017">https://bugs.webkit.org/show_bug.cgi?id=282017</a>

Reviewed by BJ Burg.

* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.0/InspectorBackendCommands.js: Removed.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.2/InspectorBackendCommands.js: Removed.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.0/InspectorBackendCommands.js: Removed.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.3/InspectorBackendCommands.js: Removed.

* Source/WebInspectorUI/Versions/Inspector-iOS-12.0.json: Removed.
* Source/WebInspectorUI/Versions/Inspector-iOS-12.2.json: Removed.
* Source/WebInspectorUI/Versions/Inspector-macOS-12.0.json: Removed.
* Source/WebInspectorUI/Versions/Inspector-macOS-12.3.json: Removed.

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI.contentLoaded):
* Source/WebInspectorUI/UserInterface/Controllers/CSSManager.js:
(WI.CSSManager.displayNameForPseudoId):
(WI.CSSManager.prototype.canForcePseudoClass):
* Source/WebInspectorUI/UserInterface/Controllers/CanvasManager.js:
(WI.CanvasManager.prototype.initializeTarget):
(WI.CanvasManager.prototype.setRecordingAutoCaptureFrameCount):
(WI.CanvasManager.supportsRecordingAutoCapture): Deleted.
* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager):
(WI.DOMDebuggerManager.supportsAllListenersBreakpoint):
(WI.DOMDebuggerManager.prototype._setEventBreakpoint):
(WI.DOMDebuggerManager.prototype._removeEventBreakpoint):
(WI.DOMDebuggerManager.prototype._setURLBreakpoint):
(WI.DOMDebuggerManager.prototype._removeURLBreakpoint):
(WI.DOMDebuggerManager.supportsEventBreakpoints): Deleted.
(WI.DOMDebuggerManager.supportsEventListenerBreakpoints): Deleted.
(WI.DOMDebuggerManager.supportsURLBreakpoints): Deleted.
(WI.DOMDebuggerManager.supportsXHRBreakpoints): Deleted.
* Source/WebInspectorUI/UserInterface/Controllers/DOMManager.js:
(WI.DOMManager.prototype.getSupportedEventNames):
* Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js:
(WI.DebuggerManager.prototype.async initializeTarget):
(WI.DebuggerManager.supportsBlackboxingBreakpointEvaluations):
(WI.DebuggerManager.prototype._setBlackboxBreakpointEvaluations):
(WI.DebuggerManager.prototype._handleEngineeringPauseForInternalScriptsSettingChanged):
* Source/WebInspectorUI/UserInterface/Controllers/DeviceSettingsManager.js:
(WI.DeviceSettingsManager.prototype.initializeTarget):
(WI.DeviceSettingsManager.prototype.overrideDeviceSetting):
(WI.DeviceSettingsManager.prototype.overrideUserAgent):
* Source/WebInspectorUI/UserInterface/Controllers/LayerTreeManager.js:
(WI.LayerTreeManager.prototype.initializeTarget):
(WI.LayerTreeManager.supportsVisibleCompositingBorders):
(WI.LayerTreeManager.prototype.set compositingBordersVisible):
* Source/WebInspectorUI/UserInterface/Controllers/RuntimeManager.js:
(WI.RuntimeManager):
(WI.RuntimeManager.preferredSavedResultPrefix):
(WI.RuntimeManager.prototype.initializeTarget):
(WI.RuntimeManager.prototype.evaluateInInspectedWindow):
(WI.RuntimeManager.supportsAwaitPromise): Deleted.
* Source/WebInspectorUI/UserInterface/Controllers/TimelineManager.js:
(WI.TimelineManager.defaultTimelineTypes):
(WI.TimelineManager.availableTimelineTypes):
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.prototype.requestEffectTarget):
* Source/WebInspectorUI/UserInterface/Models/AuditTestBase.js:
(WI.AuditTestBase.prototype.async runSetup):
* Source/WebInspectorUI/UserInterface/Models/AuditTestCase.js:
(WI.AuditTestCase.prototype.async run):
* Source/WebInspectorUI/UserInterface/Models/CPUInstrument.js:
(WI.CPUInstrument.supported): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Canvas.js:
(WI.Canvas.prototype.startRecording):
(WI.Canvas.prototype.recordingFinished):
* Source/WebInspectorUI/UserInterface/Models/DOMNode.js:
(WI.DOMNode):
* Source/WebInspectorUI/UserInterface/Models/MediaInstrument.js:
(WI.MediaInstrument.supported): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Recording.js:
(WI.Recording.fromPayload):
* Source/WebInspectorUI/UserInterface/Models/RecordingAction.js:
(WI.RecordingAction.fromPayload):
(WI.RecordingAction.prototype.async swizzle):
* Source/WebInspectorUI/UserInterface/Models/ScreenshotsInstrument.js:
(WI.ScreenshotsInstrument.supported):
* Source/WebInspectorUI/UserInterface/Protocol/Connection.js:
(InspectorBackend.Connection.prototype._dispatchEvent):
* Source/WebInspectorUI/UserInterface/Protocol/DOMObserver.js:
(WI.DOMObserver.prototype.videoLowPowerChanged): Deleted.
* Source/WebInspectorUI/UserInterface/Protocol/ScriptProfilerObserver.js:
(WI.ScriptProfilerObserver.prototype.trackingComplete):
(WI.ScriptProfilerObserver.prototype.programmaticCaptureStarted): Deleted.
(WI.ScriptProfilerObserver.prototype.programmaticCaptureStopped): Deleted.
* Source/WebInspectorUI/UserInterface/Protocol/TimelineObserver.js:
(WI.TimelineObserver.prototype.autoCaptureStarted):
(WI.TimelineObserver.prototype.programmaticCaptureStarted): Deleted.
(WI.TimelineObserver.prototype.programmaticCaptureStopped): Deleted.
* Source/WebInspectorUI/UserInterface/Views/CanvasOverviewContentView.js:
(WI.CanvasOverviewContentView):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView):
* Source/WebInspectorUI/UserInterface/Views/EventListenerSectionGroup.js:
(WI.EventListenerSectionGroup.prototype._functionTextOrLink):
* Source/WebInspectorUI/UserInterface/Views/NetworkTableContentView.js:
(WI.NetworkTableContentView):
(WI.NetworkTableContentView.prototype.closed):
(WI.NetworkTableContentView.prototype._updateOtherFiltersNavigationItemState):
(WI.NetworkTableContentView.prototype._handleOtherFiltersNavigationItemContextMenu):
* Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
(WI.SettingsTabContentView.prototype._createConsoleSettingsView):
* Source/WebInspectorUI/UserInterface/Views/SourcesNavigationSidebarPanel.js:
(WI.SourcesNavigationSidebarPanel.prototype._populateCreateBreakpointContextMenu):
* Source/WebInspectorUI/UserInterface/Views/SpreadsheetRulesStyleDetailsPanel.js:
(WI.SpreadsheetRulesStyleDetailsPanel.prototype.layout):

Canonical link: <a href="https://commits.webkit.org/290243@main">https://commits.webkit.org/290243@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/845dfcb53fc1d3bed87fa3b285975e7b986aaf8e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88628 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8147 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43073 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93590 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39382 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8534 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16331 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68332 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26034 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91630 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6526 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80152 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48696 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6290 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/34562 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/embedded-content/the-iframe-element/iframe-loading-lazy-multiple-queued-navigations.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38490 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76657 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35461 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95428 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15803 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11568 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77197 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16059 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76011 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76473 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19011 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20861 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19284 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8842 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15819 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21127 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15560 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19009 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17342 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg commented on Source/WebInspectorUI/UserInterface/Controllers/DeviceSettingsManager.js:
```diff
@@ -42,15 +42,11 @@ WI.DeviceSettingsManager = class DeviceSettingsManager extends WI.Object
 
-        // COMPATIBILITY (iOS 12.2): Page.overrideUserAgent did not exist.
-        if (target.hasCommand("Page.overrideUserAgent") && this._overridenDeviceUserAgent)
+        if (this._overridenDeviceUserAgent)
```
Existing, but `_overriden` is misspelled :|

burg commented on Source/WebInspectorUI/UserInterface/Controllers/RuntimeManager.js:
```diff
@@ -68,7 +62,7 @@ WI.RuntimeManager = class RuntimeManager extends WI.Object
             target.RuntimeAgent.enableControlFlowProfiler();
 
-        // COMPATIBILITY (iOS 12.2): Runtime.setSavedResultAlias did not exist.
+        // COMPATIBILITY (iOS 13.0): Runtime.setSavedResultAlias did not exist.
```
thanks for updating the comment

burg commented on Source/WebInspectorUI/UserInterface/Models/DOMNode.js:
```diff
@@ -64,13 +64,8 @@ WI.DOMNode = class DOMNode extends WI.Object
 
         this._frame = null;
 
-        // COMPATIBILITY (iOS 12.2): DOM.Node.frameId was changed to represent the owner frame, not the content frame.
```
Yeah, not gonna want to support this going forward :)

burg reviewed (approved):
Thanks!

dcrousso committed:
Web Inspector: drop support for iOS 12.* and macOS 12.*
https://bugs.webkit.org/show_bug.cgi?id=282017

Reviewed by BJ Burg.

* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.0/InspectorBackendCommands.js: Removed.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/iOS/12.2/InspectorBackendCommands.js: Removed.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.0/InspectorBackendCommands.js: Removed.
* Source/WebInspectorUI/UserInterface/Protocol/Legacy/macOS/12.3/InspectorBackendCommands.js: Removed.

* Source/WebInspectorUI/Versions/Inspector-iOS-12.0.json: Removed.
* Source/WebInspectorUI/Versions/Inspector-iOS-12.2.json: Removed.
* Source/WebInspectorUI/Versions/Inspector-macOS-12.0.json: Removed.
* Source/WebInspectorUI/Versions/Inspector-macOS-12.3.json: Removed.

* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Base/Main.js:
(WI.contentLoaded):
* Source/WebInspectorUI/UserInterface/Controllers/CSSManager.js:
(WI.CSSManager.displayNameForPseudoId):
(WI.CSSManager.prototype.canForcePseudoClass):
* Source/WebInspectorUI/UserInterface/Controllers/CanvasManager.js:
(WI.CanvasManager.prototype.initializeTarget):
(WI.CanvasManager.prototype.setRecordingAutoCaptureFrameCount):
(WI.CanvasManager.supportsRecordingAutoCapture): Deleted.
* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager):
(WI.DOMDebuggerManager.supportsAllListenersBreakpoint):
(WI.DOMDebuggerManager.prototype._setEventBreakpoint):
(WI.DOMDebuggerManager.prototype._removeEventBreakpoint):
(WI.DOMDebuggerManager.prototype._setURLBreakpoint):
(WI.DOMDebuggerManager.prototype._removeURLBreakpoint):
(WI.DOMDebuggerManager.supportsEventBreakpoints): Deleted.
(WI.DOMDebuggerManager.supportsEventListenerBreakpoints): Deleted.
(WI.DOMDebuggerManager.supportsURLBreakpoints): Deleted.
(WI.DOMDebuggerManager.supportsXHRBreakpoints): Deleted.
* Source/WebInspectorUI/UserInterface/Controllers/DOMManager.js:
(WI.DOMManager.prototype.getSupportedEventNames):
* Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js:
(WI.DebuggerManager.prototype.async initializeTarget):
(WI.DebuggerManager.supportsBlackboxingBreakpointEvaluations):
(WI.DebuggerManager.prototype._setBlackboxBreakpointEvaluations):
(WI.DebuggerManager.prototype._handleEngineeringPauseForInternalScriptsSettingChanged):
* Source/WebInspectorUI/UserInterface/Controllers/DeviceSettingsManager.js:
(WI.DeviceSettingsManager.prototype.initializeTarget):
(WI.DeviceSettingsManager.prototype.overrideDeviceSetting):
(WI.DeviceSettingsManager.prototype.overrideUserAgent):
* Source/WebInspectorUI/UserInterface/Controllers/LayerTreeManager.js:
(WI.LayerTreeManager.prototype.initializeTarget):
(WI.LayerTreeManager.supportsVisibleCompositingBorders):
(WI.LayerTreeManager.prototype.set compositingBordersVisible):
* Source/WebInspectorUI/UserInterface/Controllers/RuntimeManager.js:
(WI.RuntimeManager):
(WI.RuntimeManager.preferredSavedResultPrefix):
(WI.RuntimeManager.prototype.initializeTarget):
(WI.RuntimeManager.prototype.evaluateInInspectedWindow):
(WI.RuntimeManager.supportsAwaitPromise): Deleted.
* Source/WebInspectorUI/UserInterface/Controllers/TimelineManager.js:
(WI.TimelineManager.defaultTimelineTypes):
(WI.TimelineManager.availableTimelineTypes):
* Source/WebInspectorUI/UserInterface/Models/Animation.js:
(WI.Animation.prototype.requestEffectTarget):
* Source/WebInspectorUI/UserInterface/Models/AuditTestBase.js:
(WI.AuditTestBase.prototype.async runSetup):
* Source/WebInspectorUI/UserInterface/Models/AuditTestCase.js:
(WI.AuditTestCase.prototype.async run):
* Source/WebInspectorUI/UserInterface/Models/CPUInstrument.js:
(WI.CPUInstrument.supported): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Canvas.js:
(WI.Canvas.prototype.startRecording):
(WI.Canvas.prototype.recordingFinished):
* Source/WebInspectorUI/UserInterface/Models/DOMNode.js:
(WI.DOMNode):
* Source/WebInspectorUI/UserInterface/Models/MediaInstrument.js:
(WI.MediaInstrument.supported): Deleted.
* Source/WebInspectorUI/UserInterface/Models/Recording.js:
(WI.Recording.fromPayload):
* Source/WebInspectorUI/UserInterface/Models/RecordingAction.js:
(WI.RecordingAction.fromPayload):
(WI.RecordingAction.prototype.async swizzle):
* Source/WebInspectorUI/UserInterface/Models/ScreenshotsInstrument.js:
(WI.ScreenshotsInstrument.supported):
* Source/WebInspectorUI/UserInterface/Protocol/Connection.js:
(InspectorBackend.Connection.prototype._dispatchEvent):
* Source/WebInspectorUI/UserInterface/Protocol/DOMObserver.js:
(WI.DOMObserver.prototype.videoLowPowerChanged): Deleted.
* Source/WebInspectorUI/UserInterface/Protocol/ScriptProfilerObserver.js:
(WI.ScriptProfilerObserver.prototype.trackingComplete):
(WI.ScriptProfilerObserver.prototype.programmaticCaptureStarted): Deleted.
(WI.ScriptProfilerObserver.prototype.programmaticCaptureStopped): Deleted.
* Source/WebInspectorUI/UserInterface/Protocol/TimelineObserver.js:
(WI.TimelineObserver.prototype.autoCaptureStarted):
(WI.TimelineObserver.prototype.programmaticCaptureStarted): Deleted.
(WI.TimelineObserver.prototype.programmaticCaptureStopped): Deleted.
* Source/WebInspectorUI/UserInterface/Views/CanvasOverviewContentView.js:
(WI.CanvasOverviewContentView):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeContentView.js:
(WI.DOMTreeContentView):
* Source/WebInspectorUI/UserInterface/Views/EventListenerSectionGroup.js:
(WI.EventListenerSectionGroup.prototype._functionTextOrLink):
* Source/WebInspectorUI/UserInterface/Views/NetworkTableContentView.js:
(WI.NetworkTableContentView):
(WI.NetworkTableContentView.prototype.closed):
(WI.NetworkTableContentView.prototype._updateOtherFiltersNavigationItemState):
(WI.NetworkTableContentView.prototype._handleOtherFiltersNavigationItemContextMenu):
* Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
(WI.SettingsTabContentView.prototype._createConsoleSettingsView):
* Source/WebInspectorUI/UserInterface/Views/SourcesNavigationSidebarPanel.js:
(WI.SourcesNavigationSidebarPanel.prototype._populateCreateBreakpointContextMenu):
* Source/WebInspectorUI/UserInterface/Views/SpreadsheetRulesStyleDetailsPanel.js:
(WI.SpreadsheetRulesStyleDetailsPanel.prototype.layout):

Canonical link: https://commits.webkit.org/290243@main
